File: Practica_4/civitasJava/src/civitas/SorpresaIrCarcel.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas; 
 
import java.util.ArrayList;

/**
 *
 * @author mariolopezgonzalez
 */
public class SorpresaIrCarcel extends Sorpresa {

    /*
    Constructor con parámetros
    Crea una sorpresa de tipo IRCARCEL
    Parametros:
      Recibe como parámetros los valores de los atributos: tablero
     */
    SorpresaIrCarcel(Tablero tablero) {
        super("Vas a la carcel. Muévete a la casilla Cárcel", -1, tablero);
    }

    /*
    Método aplicarAJugador()
    Parámetros:
      actual: indice del jugador actual
      todos: array que contiene los jugadores
    Efecto:
       Envia al jugador actual a la cárcel
     */
    @Override
    void aplicarAJugador(int actual, ArrayList<Jugador> todos) {
        if (jugadorCorrecto(actual, todos)) {
            informe(actual, todos);
            todos.get(actual).encarcelar(getTablero().getCarcel());
        }
       
    }

    @Override
    public String toString() {
        return "SorpresaIrCarcel{ "+ super.getTexto() + ", tablero=" + getTablero() + '}';
    }
    
    
}

File: Practica_4/civitasRuby/lib/sorpresa_ir_carcel.rb
#encoding:utf-8

module Civitas
  class Sorpresa_ir_carcel < Sorpresa
    def initialize(tablero)
      super("Vas a la carcel. Muevéte a la casilla Carcel", -1, tablero)
    end 

    # Método aplicarAJugador (ir Carcel)
    # Parámetros:
    #   actual: indice del jugador actual
    #   todos: array que contiene los jugadores
    # Efecto:
    #   Envia al jugador actual a la cárcel
    def aplicar_a_jugador(actual, todos)
      if jugador_correcto(actual, todos)
        informe(actual, todos)
        todos[actual].encarcelar(tablero.num_casilla_carcel())
      end
    end

    def to_s
      return "Sorpresa Ir Carcel: Texto: #{texto}, Tablero: #{tablero}"
    end

    public_class_method :new
    
  end
end
